src/ansys/dynamicreporting/core/adr_report.py

```diff
@@ -171,6 +171,332 @@ def get_url(self, filter: str = "") -> str:
+                        // use <iframe> instead
+                        const iframeEle = document.createElement('iframe');
+                        iframeEle.src = reportPath;
+                        iframeEle.width = width;
+                        iframeEle.height = height;
+                        return this.appendChild(iframeEle);
+                    }}
+                }}
+            }}
+
+            customElements.define("adr-report", ReportFetchComponent);
+        """
+        return component_logic
+
+    def get_report_component(
+        self,
+        prefix: str = "",
+        filter: str = "",
+        style_path: str = "",
+        width: int = 1000,
+        height: int = 800,
+    ) -> str:
+        """
+        A HTML code of the web component for report fetching. By default, the web
+        component uses iframe to embed the report. If users have provided additional
+        configuration settings on their application server or on another proxy server,
+        the web component will use fetch API to embed the report directly in the
+        application.
+
+        Parameters
+        ----------
+        prefix : str, optional
+            A user defined key in the server to reroute and fetch the report from ADR server. If not provided,
+            the web component will use the default iframe to embed the report in the application.
+        filter : str, optional
+            Query string for filtering. The default is ``""``. The syntax corresponds
+            to the syntax for Ansys Dynamic Reporting. For more information, see
+            _Query Expressions in the documentation for Ansys Dynamic Reporting.
+        style_path: str, optional
+            The hosting app's stylesheet path. The default is ``""``. The syntax is used to overwrite report
+            styling using an external CSS file.
+        width : int, optional
+            Width of the iframe if the web component uses <iframe> to embed report. The default is ``1000``.
+        height : int, optional
+            Height of the iframe if the web component uses <iframe> to embed report. The default is ``800``.
+
+        Returns
+        -------
+        str
+            The web component HTML code (as string) that will get embedded in the HTML page
+
+        Examples
+        --------
+        ::
+
+            import ansys.dynamicreporting.core as adr
+            adr_service = adr.Service(ansys_installation = r'C:\\Program Files\\ANSYS Inc\\v232')
+            ret = adr_service.connect()
+            my_report = adr_service.get_report(report_name = 'Top report')
+            my_report.get_report_component()
+        """
+        # fetch method using predefined prefix rules in the proxy server OR using traditional <iframe>
+        # add host-style-path attribute if specified (can only work when prefix is provided)
+        host_style_path = f'host-style-path="{style_path}"' if style_path else ""
+        fetchMethod = (

```

Comment:
snake_case here

Reply:
Thanks for spotting this, writing JS for so long have got me default to camelCase

Reply:
@phchen95 lol no worries, I can totally relate to this